# Generated by Django 4.2.3 on 2024-02-07 19:36

from django.conf import settings
import django.contrib.gis.db.models.fields
from django.db import migrations, models
import django.db.models.deletion


class Migration(migrations.Migration):

    dependencies = [
        ('rescuenetApp', '0005_hospital_school'),
    ]

    operations = [
        migrations.CreateModel(
            name='Disaster',
            fields=[
                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('name', models.CharField(max_length=255)),
                ('location', django.contrib.gis.db.models.fields.PointField(srid=4326)),
                ('address', models.CharField(max_length=255)),
                ('disaster_type', models.CharField(max_length=255)),
                ('timestamp', models.DateTimeField(auto_now_add=True)),
                ('details', models.TextField()),
            ],
        ),
        migrations.CreateModel(
            name='AgencyGroup',
            fields=[
                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('name', models.CharField(max_length=255)),
                ('dissolution_time', models.DateTimeField()),
                ('agencies', models.ManyToManyField(to=settings.AUTH_USER_MODEL)),
                ('disaster', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to='rescuenetApp.disaster')),
            ],
        ),
    ]
